ts41/query-result.d.ts

```diff
@@ -0,0 +1,70 @@
+import {LastArrayElement} from '..';
+import {Split} from './split';
+import {Trim} from './trim';
+
+/** Strips a given modifier from a string. */
+type StripModifier<V extends string, M extends string> = V extends `${infer L}${M}${infer A}` ? L : V;
+
+/** Strips CSS modifiers from a string. */
+type StripModifiers<V extends string> = StripModifier<StripModifier<StripModifier<StripModifier<V, '.'>, '#'>, '['>, ':'>;
+
+/** Takes the last element after the given token. */
+type LastArrayElementAfterToken<V extends string, T extends string> = StripModifiers<LastArrayElement<Split<Trim<V>, T>>>;
+
+/** Extracts the last element from a CSS selector string. */
+type GetLastElementName<V extends string> = LastArrayElementAfterToken<LastArrayElementAfterToken<V, ' '>, '>'>;
+
+/** Returns an array of element names with stripped modifiers. */
+type GetEachElementName<V, L extends string[] = []> =
```

Comment:
This type is a little confusing to me. Again it feels like the `[string]` check isn't necessary since it's covered by `[string, ...infer Tail]` but I could be wrong. Maybe some examples in the jsdoc could help either way. I know not exposed but I suspect this implementation will be used as a reference.